src/name_core.jl

```diff
@@ -0,0 +1,97 @@
+
+"""
+    name2dim(dimnames, [name])
+
+For `dimnames` being a tuple of dimnames (symbols) for dimenensions.
+If called with just the tuple,
+returns a named tuple, with each name maps to a dimension.
+e.g `name2dim((:a, :b)) == (a=1, b=2)`.
+
+If the second `name` argument is given, them the dimension corresponding to that `name`,
+is returned.
+e.g. `name2dim((:a, :b), :b) == 2`
+If that `name` is not found then `0` is returned.
+"""
+function name2dim(dimnames::Tuple)
```

Comment:
```suggestion
function name2dim(dimnames::NTuple{N, Symbol}) where N
```
And then you could get rid of the second line of the function

Reply:
Also applies to `default_inds`